fe/fe-core/src/main/java/com/starrocks/catalog/BaseTableInfo.java

```diff
@@ -162,92 +179,62 @@ public Table getTableChecked() {
         throw MaterializedViewExceptions.reportBaseTableNotExists(this);
     }
 
+    /**
+     * Get {@link Table} from {@code BaseTableInfo} which it can be an OlapTable or ExternalTable.
+     * </p>
+     * NOTE: Prefer {@link BaseTableInfo#getTableChecked()} or {@link BaseTableInfo#mayGetTable()}
+     * because this method may return null if base table has already dropped or schema changed.
+     * </p>
+     * @return {@link Table} if BaseTableInfo is health otherwise null will be returned.
+     */
     @Deprecated
```

Comment:
when to make this api private?

Reply:
later... because this will change too much codes.